src/SerilogWeb.Classic/Classic/WebApiExceptionLogger.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web;
+using System.Web.Http;
+using System.Web.Http.ExceptionHandling;
+
+namespace SerilogWeb.Classic
+{
+    /// <summary>
+    /// Add the following line in Global.asax.cs
+    /// to log WebApi faults
+    /// <code>
+    /// protected void Application_Start()
+    ///   {
+    ///       GlobalConfiguration.Configure(SerilogWeb.Classic.WebApiExceptionLogger.Register);
+    ///   }
+    /// </code>
+    /// </summary>
+    public class WebApiExceptionLogger : IExceptionLogger
+    {
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public static Action<HttpConfiguration> Register
+            => c => c.Services.Add(typeof (IExceptionLogger),
+                new WebApiExceptionLogger());
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
```

Comment:
I pushed it up last night. What is the process of publishing to nuget? Does serilog have is own account?

-----Original Message-----
From: "Ian Yates" <notifications@github.com>
Sent: ‎15/‎01/‎2016 10:39 AM
To: "serilog-web/classic" <classic@noreply.github.com>
Cc: "Chui Tey" <teyc@cognoware.com>
Subject: Re: [classic] Log webapi exceptions (#24)

In src/SerilogWeb.Classic/Classic/WebApiExceptionLogger.cs:
> +    {
> +
> +        /// <summary>
> +        /// 
> +        /// </summary>
> +        public static Action<HttpConfiguration> Register
> +            => c => c.Services.Add(typeof (IExceptionLogger),
> +                new WebApiExceptionLogger());
> +
> +        /// <summary>
> +        /// 
> +        /// </summary>
> +        /// <param name="context"></param>
> +        /// <param name="cancellationToken"></param>
> +        /// <returns></returns>
> +        public Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
No worries. I'll let you put this stuff into the new repo @nblumhardt set up. (and thanks Nick for making me a contributor!)
—
Reply to this email directly or view it on GitHub.